### DISCUSSION TITLE
Drop shadow parameter for getFilePreview

### DISCUSSION THREAD

**Discussion by Meldiron** [appwrite/appwrite#2409] [Ideas]
Currently, if I use the rotation parameter in `getFilePreview`, there is no way for me to achieve drop shadow. The closest I can get to is:

![image](https://user-images.githubusercontent.com/19310830/141441927-e96831b1-674e-47f6-8d07-eda8e76053f2.png)

I think there should be additional parameters for defining drop shadow, just like there is for borders. This will allow us to use drop shadows on rotated images. We should be able to define:

- Type (inner/outer)
- Color (hex)
- Opacity (float 0-1)
- size & position

On the other hand, that might be too hard to configure properly, so we could just add one parameter `shadow` that could be one of `inner, outer-small, outer-medium, outer-big`. This will give fewer options but they are easy to understand and can give Appwrite space to create **the best** drop shadows possible.

For example, to make drop shadows feel natural, there should be 2 layers of shadow, as described in the screenshot below. That would be extremely hard to configure on my own, but "outer-medium" could have that defined on the Appwrite side.

![CleanShot 2021-11-12 at 10 20 34@2x](https://user-images.githubusercontent.com/19310830/141443022-9d312c86-7917-4c46-bbb1-11fd15dbcb08.png)

---

**What do you think?** Which approach is better? Can you see any other way to implement it? Do we even need drop shadow in `getFilePreview`?

**Comment by eldadfux** [comment]
For reference, this is the Cloudinary implementation: https://cloudinary.com/cookbook/add_a_shadow_effect_to_an_image
My main concern is that this might introduce a lot of new param to the API in order to control the shadow display. @Meldiron how would you imagine the API signature for this functionality?
